### DISCUSSION TITLE
Analytical Table: Disable all the checkbox including 'select all' checkbox

### DISCUSSION THREAD

**Discussion by shivamdz** [UI5/webcomponents-react#3076] [Q&A] [answered]
Hi Everyone,

We have a requirement where we need to disable all the checkbox, including the 'select all' checkbox present in the header.
I tried using 'useRowDisableSelection' which disables all the checkbox of the rows but it remove the header 'select all' checkbox. 
My requirement is to show 'select all' checkbox also but in disabled state.

Any suggestion on how we can do it?

Thanks,
Shivam

**Comment by Lukas742** [comment, marked as answer]
Hi @shivamdz 

The easiest way would be to use both the useRowDisableSelection plugin hook and your own hook. --> [codeSandbox](https://codesandbox.io/s/defaulttable-forked-h2evjj?file=/src/App.js)

The most performant way would be to copy the [useRowDisableSelection](https://github.com/SAP/ui5-webcomponents-react/blob/main/packages/main/src/components/AnalyticalTable/pluginHooks/useRowDisableSelection.tsx) hook and edit it on your side to fit your use case, as this way you would only need to register one custom hook. You can probably reuse parts of the implementation of the codeSandbox above.
